services/web/client/source/class/osparc/component/widget/newStudy/WithToken.js

```diff
@@ -0,0 +1,125 @@
+/* ************************************************************************
+
+   osparc - the simcore frontend
+
+   https://osparc.io
+
+   Copyright:
+     2019 IT'IS Foundation, https://itis.swiss
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+
+   Authors:
+     * Odei Maiz (odeimaiz)
+
+************************************************************************ */
+
+/**
+ * Widget that provides the form for creating a new study for a given token
+ *
+ * After doing some token validation the following data event is fired:
+ * <pre class='javascript'>
+ *   {
+ *     studyId: studyId
+ *   };
+ * </pre>
+ *
+ * *Example*
+ *
+ * Here is a little example of how to use the widget.
+ *
+ * <pre class='javascript'>
+ *   let newStudyWithToken = new osparc.component.widget.newStudy.WithToken();
+ *   this.getRoot().add(newStudyWithToken);
+ * </pre>
+ */
+
+qx.Class.define("osparc.component.widget.newStudy.WithToken", {
```

Comment:
If we were making a refactor on the namespace tree, I would get rid of the widget folder because it is quite redundant. Everything inside ``component`` are widgets, or should be... These components are forms right? So maybe we should put them under ``osparc.component.form.study.*``